Use c++11 for size test

lucylq opened the pull request.

swolchok reviewed (approved):
I don't love the `string_view` changes from an aesthetic perspective, but they do look necessary.

lucylq committed:
Use c++11 for size test (#3509)

Summary: Pull Request resolved: https://github.com/pytorch/executorch/pull/3509

Reviewed By: mergennachin, swolchok

Differential Revision: D57003591

Pulled By: lucylq
